In [1]:
use mavenfuzzyfactory;

Commands completed successfully

Total execution time: 00:00:00.004

In [2]:
show TABLES

(6 row(s) affected)

Total execution time: 00:00:01.029

Tables_in_mavenfuzzyfactory order_item_refunds order_items orders products website_pageviews website_sessions

In [3]:
select * from website_sessions limit 5

(5 row(s) affected)

Total execution time: 00:00:01.009

website_session_id created_at user_id is_repeat_session utm_source utm_campaign utm_content device_type http_referer 1 2012-03-19 08:04:16 1 0 gsearch nonbrand g_ad_1 mobile https://www.gsearch.com 2 2012-03-19 08:16:49 2 0 gsearch nonbrand g_ad_1 desktop https://www.gsearch.com 3 2012-03-19 08:26:55 3 0 gsearch nonbrand g_ad_1 desktop https://www.gsearch.com 4 2012-03-19 08:37:33 4 0 gsearch nonbrand g_ad_1 desktop https://www.gsearch.com 5 2012-03-19 09:00:55 5 0 gsearch nonbrand g_ad_1 mobile https://www.gsearch.com

In [4]:
select * from website_pageviews limit 5

(5 row(s) affected)

Total execution time: 00:00:01.006

website_pageview_id created_at website_session_id pageview_url 1 2012-03-19 08:04:16 1 /home 2 2012-03-19 08:16:49 2 /home 3 2012-03-19 08:26:55 3 /home 4 2012-03-19 08:37:33 4 /home 5 2012-03-19 09:00:55 5 /home

In [5]:
select * from orders limit 5

(5 row(s) affected)

Total execution time: 00:00:01.008

order_id created_at website_session_id user_id primary_product_id items_purchased price_usd cogs_usd 1 2012-03-19 10:42:46 20 20 1 1 49.99 19.49 2 2012-03-19 19:27:37 104 104 1 1 49.99 19.49 3 2012-03-20 06:44:45 147 147 1 1 49.99 19.49 4 2012-03-20 09:41:45 160 160 1 1 49.99 19.49 5 2012-03-20 11:28:15 177 177 1 1 49.99 19.49

In [6]:
select * from order_items limit 5

(5 row(s) affected)

Total execution time: 00:00:01.003

order_item_id created_at order_id product_id is_primary_item price_usd cogs_usd 1 2012-03-19 10:42:46 1 1 1 49.99 19.49 2 2012-03-19 19:27:37 2 1 1 49.99 19.49 3 2012-03-20 06:44:45 3 1 1 49.99 19.49 4 2012-03-20 09:41:45 4 1 1 49.99 19.49 5 2012-03-20 11:28:15 5 1 1 49.99 19.49

**TASK:** We've been live for almost a month now and we're <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">starting to generate sales.&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">Where the bulk of our website sessions are coming f</span><span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">rom, through yesterday?&nbsp;</span>  <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">I'd like to see a breakdown by UTM source, campaign&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">and referring domain if possible.&nbsp;</span>

In [7]:
select utm_source, utm_campaign, http_referer, count(DISTINCT website_session_id) as no_sessions
from website_sessions WHERE created_at < '2012-04-12' 
GROUP by utm_source, utm_campaign, http_referer
ORDER by no_sessions desc

(6 row(s) affected)

Total execution time: 00:00:01.230

utm_source utm_campaign http_referer no_sessions gsearch nonbrand https://www.gsearch.com 3613 NULL NULL NULL 28 NULL NULL https://www.gsearch.com 27 gsearch brand https://www.gsearch.com 26 NULL NULL https://www.bsearch.com 7 bsearch brand https://www.bsearch.com 7

**Answer:** <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);"> Most Traffic on website was from GSearch from non branded keywords.</span>

**TASK:** Sounds like search nonbrand is our major traffic source, but <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">we need to understand if those sessions are driving sales.&nbsp;</span> Calculate the conversion rate (VR) from <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">session to order? Based on what we're paying for clicks,&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">we'll need a CVR of at least 4% to make the numbers work.&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">If we're much lower, we'll need to reduce bids. If we're&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">higher, we can increase bids to drive more volume.</span>

In [8]:
select * FROM orders limit 2

(2 row(s) affected)

Total execution time: 00:00:01.007

order_id created_at website_session_id user_id primary_product_id items_purchased price_usd cogs_usd 1 2012-03-19 10:42:46 20 20 1 1 49.99 19.49 2 2012-03-19 19:27:37 104 104 1 1 49.99 19.49

In [9]:
select * from website_sessions limit 2

(2 row(s) affected)

Total execution time: 00:00:01.006

website_session_id created_at user_id is_repeat_session utm_source utm_campaign utm_content device_type http_referer 1 2012-03-19 08:04:16 1 0 gsearch nonbrand g_ad_1 mobile https://www.gsearch.com 2 2012-03-19 08:16:49 2 0 gsearch nonbrand g_ad_1 desktop https://www.gsearch.com

In [10]:
select count(DISTINCT website_sessions.website_session_id) as sessions, count(DISTINCT orders.order_id) as orders, 
count(DISTINCT orders.order_id)/count(DISTINCT website_sessions.website_session_id) as session_to_order_conv_rt
FROM website_sessions
LEFT JOIN orders
ON website_sessions.website_session_id = orders.website_session_id
WHERE website_sessions.created_at < '2012-04-14'
AND website_sessions.utm_source = 'gsearch' 
AND website_sessions.utm_campaign = 'nonbrand'

(1 row(s) affected)

Total execution time: 00:00:01.286

sessions orders session_to_order_conv_rt 3895 112 0.0288

**TASK - Gsearch volume trends:** Can you pull gsearch nonbrand trended session volume, by <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">week, to see if the bid changes have caused volume to drop&nbsp;</span>  <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">at all?</span>

In [11]:
select count(DISTINCT(website_session_id)),  
-- YEAR(created_at), 
-- WEEK(created_at),
DATE(created_at)
FROM website_sessions
where created_at < '2012-05-12'
AND utm_source = 'gsearch' 
AND utm_campaign = 'nonbrand'
GROUP BY YEAR(created_at), WEEK(created_at)

(8 row(s) affected)

Total execution time: 00:00:01.247

count(DISTINCT(website_session_id)) DATE(created_at) 896 2012-03-19 956 2012-03-25 1152 2012-04-01 983 2012-04-08 621 2012-04-15 594 2012-04-22 681 2012-04-29 651 2012-05-06

**ANSWER:** Drop in sessions in past two weeks from 681 to 651. Indicating changes in bid optimisation is related to drop in sessions.

**SUBJECT**: **search device-level performance:** Could you pull conversion rates from session to order, by <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">device type?</span>

If desktop performance is better than on mobile we may be <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">able to bid up for desktop specifically to get more volume?</span>

In [12]:
SELECT  website_sessions.device_type, count(DISTINCT website_sessions.website_session_id) as sessions, count( DISTINCT orders.order_id) as orders, 
count(DISTINCT orders.order_id)/count(DISTINCT website_sessions.website_session_id) order_to_session
FROM website_sessions
LEFT JOIN orders 
ON website_sessions.website_session_id = orders.website_session_id
WHERE website_sessions.created_at < '2012-05-11'
AND website_sessions.utm_source = 'gsearch' 
AND website_sessions.utm_campaign = 'nonbrand'
GROUP BY website_sessions.device_type

(2 row(s) affected)

Total execution time: 00:00:01.269

device_type sessions orders order_to_session desktop 3911 146 0.0373 mobile 2492 24 0.0096

**Subject: Gsearch device-level trends:** After your device-level analysis of conversion rates, we <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">realized desktop was doing well, so we bid our gsearch&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">nonbrand desktop campaigns up on 2012-05-19.&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">Could you pull weekly trends for both desktop and mobile&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">so we can see the impact on volume?&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: var(--vscode-foreground);">You can use 2012-04-15 until the bid change as a baseline.</span>

In [13]:
SELECT DATE(website_sessions.created_at) start_of_week,
COUNT(DISTINCT CASE when website_sessions.device_type='desktop' then website_sessions.website_session_id else null end) as desktop,
COUNT(DISTINCT CASE when website_sessions.device_type='mobile' then website_sessions.website_session_id else null end) as mobile
-- count(DISTINCT website_sessions.website_session_id) as sessions, count( DISTINCT orders.order_id) as orders,
-- count(DISTINCT orders.order_id)/count(DISTINCT website_sessions.website_session_id) order_to_session
FROM website_sessions
LEFT JOIN orders 
ON website_sessions.website_session_id = orders.website_session_id
WHERE website_sessions.created_at < '2012-06-09' 
AND website_sessions.created_at > '2012-04-15'
AND website_sessions.utm_source = 'gsearch' 
AND website_sessions.utm_campaign = 'nonbrand'
GROUP BY WEEK(website_sessions.created_at)

(8 row(s) affected)

Total execution time: 00:00:01.270

start_of_week desktop mobile 2012-04-15 383 238 2012-04-22 360 234 2012-04-29 425 256 2012-05-06 430 282 2012-05-13 403 214 2012-05-20 661 190 2012-05-27 585 183 2012-06-03 582 157